docs/zoneinfo.rst

```diff
@@ -0,0 +1,370 @@
+    2020-11-01 01:00:00-07:00
+
+Data sources
+------------
+
+The ``zoneinfo`` module does not directly provide time zone data, and instead
+pulls time zone information from the system time zone database or the
+first-party PyPI package ``tzdata``, if available. Some systems, including
+notably Windows systems, do not have an IANA database available, and so for
+projects targeting cross-platform compatibility that require time zone data, it
+is recommended to declare a dependency on ``tzdata``.
+
+.. _zoneinfo_data_configuration:
+
+Configuring the data sources
+****************************
+
+When ``ZoneInfo(key)`` is called, the constructor first searches the
+directories specified in :data:`TZPATH` a match to ``key``, and on failure
+looks for a match in the ``tzdata`` package. This behavior can be configured
+in three ways:
+
+1. The default :data:`TZPATH` when not otherwise specified can be configured at
+   compile time.
+2. :data:`TZPATH` can be configured using an environment variable.
+3. At runtime, the search path can be manipulated using the
+   :func:`reset_tzpath` function.
+
+Compile-time configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The default :data:`TZPATH` includes several common deployment locations for the
+time zone database (except on Windows, where there are no "well-known"
+locations for time zone data). Downstream distributors and those building
+Python from source who know where their system time zone data is deployed may
+change the default time zone path by specifying the compile-time option
+``PYTHONTZPATHDEFAULT``, which should be a string delimited by
+:data:`os.pathsep`.
+
+Environment configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When initializing :data:`TZPATH` (either at import time or whenever
+:func:`reset_tzpath` is called with no arguments), the ``zoneinfo`` module will
+use the environment variable ``PYTHONTZPATH``, if it exists, to set the search
+path.
+
+.. envvar:: PYTHONTZPATH
+
+    This is an :data:`os.pathsep`-separated string containing the time zone
+    search path to use. It must consist of only absolute rather than relative
+    paths. Relative components specified in ``PYTHONTZPATH`` will not be used,
+    but otherwise the behavior when a relative path is specified is
+    implementation-defined; CPython will raise :exc:`InvalidTZPathWarning`, but
+    other implementations are free to silently ignore the erroneous component
+    or raise an exception.
+
+To set the system to ignore the system data and use the ``tzdata`` package
+instead, set ``PYTHONTZPATH=""``.
+
+Runtime configuration
+^^^^^^^^^^^^^^^^^^^^^
+
+The TZ search path can also be configured at runtime using the
+:func:`reset_tzpath` function. This is generally not an advisable operation,
+though it is reasonable to use it in test functions that require the use of a
+specific time zone path (or require disabling access to the system time zones).
+
+
+The ``ZoneInfo`` class
+----------------------
+
+.. class:: ZoneInfo(key)
```

Comment:
Why is it `key` and not some `tz_name`? Also, it'd be a great case for a positional-only arg.

Reply:
It's called `key` because these are properly primary keys to the time zone database, not necessarily "names" of the time zone.

It would normally be a reasonable positional-only argument, except that in `ZoneInfo.from_file` it's actually *keyword only*, and this also matches the name of the `ZoneInfo.key` attribute, so the actual name of the argument is pretty clearly established.

Also, it's a bit late to change this, since the signature and nomenclature is baked into the PEP.

Reply:
Ah, got it